shared/src/msi-installer/shared/EnvironmentVariables.wxs

```diff
@@ -7,7 +7,11 @@
     <ComponentGroup Id="Shared.EnvironmentVariables.Machine" Directory="INSTALLFOLDER">
       <Component Id="EnvironmentVariablesShared" Guid="{C314A305-9C24-4E46-9ECF-E5EEA703BDEA}" Win64="$(var.Win64)">
         <CreateFolder/>
-        <Environment Id="DD_NATIVELOADER_CONFIGFILE" Name="DD_NATIVELOADER_CONFIGFILE" Action="set" Permanent="no" System="yes" Value="[INSTALLFOLDER]loader.conf" Part="all" />
+        <?if $(var.Win64) = yes ?>
```

Comment:
I suspected that this would not work when 1) the 64-bit installer is installed and 2) a 32-bit process is being instrumented, and I was able to verify this issue locally. When this happens the loader emits the following log lines:
```
[2022-08-08 14:01:16.683 | debug | PId: 62804 | TId: 56360] Load: C:\Program Files\Datadog\.NET Tracer\win-x64\Datadog.Tracer.Native.dll
[2022-08-08 14:01:16.683 | warning | PId: 62804 | TId: 56360] Load: Error loading dynamic library 'C:\Program Files\Datadog\.NET Tracer\win-x64\Datadog.Tracer.Native.dll': 02F66CA0
```

This is only an issue with the MSI and specifically the 64-bit installer (which should instrument both 64-bit applications and 32-bit applications). To support this scenario, I think we'll need to do the following:
1. Install the `loader.conf` at the `[INSTALLFOLDER]` location
2. This `loader.conf` will need to include the `win-x64` and `win-x86` subdirectories in its lookup paths
3. Revert the changes to this file so the environment variable `DD_NATIVELOADER_CONFIGFILE` always points to the `[INSTALLFOLDER]loader.conf` location

The modified loader.conf should look like the following:
```
#Profiler
PROFILER;{BD1A650D-AC5D-4896-B64F-D6FA25D6B26A};win-x64;.\win-x64\Datadog.Profiler.Native.dll
PROFILER;{BD1A650D-AC5D-4896-B64F-D6FA25D6B26A};win-x86;.\win-x86\Datadog.Profiler.Native.dll

#Tracer
TRACER;{50DA5EED-F1ED-B00B-1055-5AFE55A1ADE5};win-x64;.\win-x64\Datadog.Tracer.Native.dll
TRACER;{50DA5EED-F1ED-B00B-1055-5AFE55A1ADE5};win-x86;.\win-x86\Datadog.Tracer.Native.dll
```

Reply:
I started a thread [here](https://github.com/DataDog/dd-trace-dotnet/pull/3060#discussion_r940691673) where I suggested we could move all of the 32-bit assets into the `Program Files (x86)` folder just like `Datadog.Trace.ClrProfiler.Native.dll`, since that's required to be placed there. If we did that AND we did not set `DD_NATIVELOADER_CONFIGFILE` globally, then this would also fix the issue I found since all of the 64-bit files (and the loader) would be in the same directory as the 64-bit `Datadog.Trace.ClrProfiler.Native.dll` loader and all of the 32-bit files (and the loader) would be in the same directory as the 32-bit `Datadog.Trace.ClrProfiler.Native.dll` loader. This seems to me like the cleanest solution. What do you think?

Reply:
I think having everything 32-bit in the 32-bit program folder makes a lot of sense. My only question is whether having all the managed dlls in a separate folder away from the native loaders becomes an issue then.. It breaks the "clean, unified layout" we currently have (but the MSI breaks that anyway by having two install locations).

Essentially it's a question of whether it's better to have
- x86 assets all in the same location
- simple loader conf
- tracer home cannot be inferred from native dll locations, and is in a separate location

or 
- native loader separate from x86 native dlls + loader conf
- tweaked loader conf
- tracer home _can_ be inferred from native tracer etc

I was initially thinking the first approach, but now I'm leaning to the second approach...

Also... I didn't realise we don't have _any_ 32-bit on x64 tests in CI 😬 We should have one at least! I'll try adding a smoke test separately...

Reply:
Have we considered installing _all_ the files into a single base location, either `Program Files` or `Program Files (x86)` or other, instead of splitting them up? I think we can do that now that we have the `win-x64` and `win-x86` folders, and it could make things simpler by keeping everything contained in a single path (like all the other packages). On the other hand, this may require some MSI trickery, since MSI can be finicky about what is x64 vs x86.

Related: there is a long-standing bug where we allow users to change the tracer installation path, but that only applies to the files in `Program Files`. Users _cannot_ change where the `Program Files (x86)` files are installed. We've had support cases of users not wanting to install _anything_ into `C:\`, for example, but that is currently not possible.

Reply:
> Have we considered installing all the files into a single base location, either Program Files or Program Files (x86) or other, instead of splitting them up?

MSIs aren't my strong point, so I don't know why we split them up at all (I thought we always had the `win-x64` and `winx-86` folders). I assumed it was because we register via the ID with the MSI, not using the profiler path, so I assumed the profiler had to be in the expected folder (i.e. `Program Files (X86)` for 32-bit processes)

If we can avoid it, I'm strongly in favour of not splitting them up! Especially if it solves additional issues.

Reply:
Also discussed on video chat is the possibility of using `COR_PROFILER_PATH_32` and `COR_PROFILER_PATH_64` to support both types of processes at the same time. We need to figure out which version of .NET Framework added support for these env vars.

Reply:
To close the loop here, I tested the use of the environment variables `COR_PROFILER_PATH_32` and `COR_PROFILER_PATH_64` on a machine with .NET Framework 4.6.1 and automatic instrumentation was correctly enabled. So the next course of action is to:

1. Remove the `<Class />` entries in `shared/src/msi-installer/shared/Files.wxs` so we no longer register the native loader (`Datadog.Trace.ClrProfiler.Native`) via COM
2. For the 64-bit installer, update the 32-bit native loader (`Datadog.Trace.ClrProfiler.Native`) so its `Win64="yes"`
3. Add machine-wide environment variables in `shared/src/msi-installer/shared/EnvironmentVariables.wxs` for `COR_PROFILER_PATH_32`/`COR_PROFILER_PATH_64`/`CORECLR_PROFILER_PATH_32`/`CORECLR_PROFILER_PATH_64`
4. Extra: Since we're here, maybe we should set `COR_PROFILER`/`CORECLR_PROFILER` as machine-wide environment variables so customers only need to set `COR_ENABLE_PROFILING`/`CORECLR_ENABLE_PROFILING` to enable automatic instrumentation for their non-IIS apps?

Reply:
I made all these changes in https://github.com/DataDog/dd-trace-dotnet/pull/3060/commits/802137d5733d57be0cdf3de70e3810054cbcb8ca, please double check it! 😄 

An additional change I made in that commit was to remove the `DD_NATIVELOADER_CONFIGFILE` env var we were setting, as the correct defaults will be inferred now anyway, and it was previously pointing to the wrong file when running an x86 app with x64 installer. 

**Question: As we now set the `COR_PROFILER`/`CORECLR_PROFILER` GUID as a machine-wide env var, should we _remove_ it from the IIS registry keys to avoid the duplication?**

I confirmed that with the previous setup, [our x86 smoke tests with x64 installer failed](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=107312&view=logs&j=5d046ac3-5cf7-5682-9fee-69a3387317d8&t=2da55ed7-cd6b-5596-bf4e-0812cbeafcb5) (which is good, confirms the test is working). With the new setup, [those tests pass](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=107323&view=results) 🎉.

I've also updated the description with the new final x64 layout:
![image](https://user-images.githubusercontent.com/18755388/183854079-5dfb5201-5378-4933-8eaa-62ebc637edad.png)

Reply:
Awesome!

Reply:
Those changes look good to me! We should also update our docs to remove the unnecessary COR_PROFILER/CORECLR_PROFILER GUID.

> **Question: As we now set the COR_PROFILER/CORECLR_PROFILER GUID as a machine-wide env var, should we remove it from the IIS registry keys to avoid the duplication?**

I say go for it 👍🏼

Reply:
I think it's always clearer for the user (or troubleshooter) to have all the config variables in the same place. If we are setting the IIS variables in the registry I find it always usefull to have them all together, althaough some of them could be declared globally.

Reply:
> I think it's always clearer for the user (or troubleshooter) to have all the config variables in the same place.

The trouble is we need to support IIS, windows services and ad-hoc apps. With the approach we've taken here, users _only_ need to set `COR_ENABLE_PROFILING` for the windows services or for the standalone apps, nothing else needs setting.

If we _don't_ take this approach then either:

- We require customers to set the env var paths to the profiler correctly when running windows services etc. That's potentially 4/8 env vars... I think it's better to only require the `COR_ENABLE_PROFILING`/`CORECLR_ENABLE_PROFILING` variables, as they're harder to screw up
- Duplicate config variables in multiple places, i.e. globally _and_ in IIS. There's more chances for things to go out sync then

Reply:
Updated docs: https://github.com/DataDog/documentation/pull/14980